subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/ClientForwardingTaskOperationListener.java

```diff
@@ -128,14 +128,4 @@ private static AbstractTaskResult toTaskResult(TaskInternal task, OperationFinis
         }
     }
 
-    private Set<DefaultTaskDescriptor> computeTaskDependencies(TaskInternal task) {
-        ProjectInternal rootProject = (ProjectInternal) task.getProject().getRootProject();
-        TaskExecutionGraph taskExecutionGraph = taskExecutionGraphs.get(rootProject.getIdentityPath());
-        return taskExecutionGraph.getDependencies(task).stream()
-            .map(dependency -> ((TaskInternal) dependency).getTaskIdentity())
-            .map(descriptors::get)
-            .filter(Objects::nonNull)
-            .collect(toCollection(LinkedHashSet::new));
-    }
```

Comment:
I like that we are repackaging things that come as a `Set` into another `Set` and instead use `Stream`s: this save a lot of time, since creating a new `LinkedHashSet` is very expensive.